### PR TITLE
i#4658: Fix fragment_prefix_size on AArch64.

### PR DESCRIPTION
**Pull request by abhinav92003**
`fragment_prefix_size` returns an incorrect value on AArch64 when `-shared_bb_ibt_tables` is set. To fix this, we set the default value of `-bb_prefixes` to true on AArch64.

Fixes: #4658

### CODE REVIEW

File: core/optionsx.h [hunk 1]
```diff
@@ -497,7 +497,8 @@ OPTION_COMMAND(bool, opt_memory, false, "opt_memory",
                "enable memory savings at potential loss in performance", STATIC,
                OP_PCACHE_NOP)
 
-PC_OPTION_INTERNAL(bool, bb_prefixes, "give all bbs a prefix")
+PC_OPTION_DEFAULT_INTERNAL(bool, bb_prefixes, IF_AARCH64_ELSE(true, false),
```
Comment (derekbruening) [inline core/optionsx.h:500]: This option cannot be disabled for A64, so it seems best to add an options.c rule to prevent that.

Comment (abhinav92003) [reply core/optionsx.h:500]: Done.

File: core/options.c [hunk 2]
```diff
@@ -1232,6 +1232,12 @@ static bool
 check_option_compatibility_helper(int recurse_count)
 {
     bool changed_options = false;
+#    if defined(CLIENT_INTERFACE) && defined(AARCH64)
+    if (!DYNAMO_OPTION(bb_prefixes)) {
+        dynamo_options.bb_prefixes = true;
```
Comment (derekbruening) [inline core/options.c:1237]: I think a USAGE_ERROR is called for here to let the user know, like in the other cases.

**Review by derekbruening — APPROVED** [review]

File: core/options.c (hunk 2 above)
Comment (abhinav92003) [reply core/options.c:1238]: Done.

**Pull request merged**